infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -167,15 +126,25 @@ public long size() {
      * @param newTail        global tail
      */
     public void put(ConflictTxStream conflictStream, long newTail) {
-        conflictCache.put(conflictStream, newTail);
+        if (cacheHash.size() == cacheSize) {
+            ConflictTxStreamEntry entry = cacheEntries.peek();
+            invalidateUpTo(firstAddress() + 1);
+        }
+
+        ConflictTxStreamEntry entry = new ConflictTxStreamEntry(conflictStream, newTail);
+        cacheEntries.add(entry);
+        cacheHash.put(conflictStream, entry.txVersion);
+        maxAddress = Math.max(newTail, maxAddress);
```

Comment:
You don't need to track a `maxAddress` because `updateMaxConflictAddress` is exposed and it will be called with `cache.updateMaxConflictAddress(initialToken - 1);`